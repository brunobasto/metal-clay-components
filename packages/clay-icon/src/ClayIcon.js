'use strict';

import Component from 'metal-component';
import Soy from 'metal-soy';
import { validators } from 'metal-state';

import templates from './ClayIcon.soy.js';

/**
 * Implementation of the Metal Clay Icon.
 */
class ClayIcon extends Component {}

/**
 * State definition.
 * @static
 * @type {!Object}
 */
ClayIcon.STATE = {
	/**
	 * The path to the SVG spritemap file containing the icons.
	 * @instance
	 * @memberof ClayIcon
	 * @type {?string}
	 * @default undefined
	 */
	spritemap: {
		validator: validators.string,
	},

	/**
	 * The name of the Clay SVG Icon e.g. `plus`.
	 * @instance
	 * @memberof ClayIcon
	 * @type {?string}
	 * @default undefined
	 */
	symbol: {
		validator: validators.string,
	},
};

Soy.register(ClayIcon, templates);

export { ClayIcon };
export default ClayIcon;
