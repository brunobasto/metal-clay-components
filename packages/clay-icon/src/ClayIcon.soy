{namespace ClayIcon}

/**
 * This renders an icon element based on the Clay Icon definition.
 */
{template .render}
	{@param? elementClasses: string}
	{@param? spritemap: string}
	{@param? symbol: string}

	{if isNonnull($symbol) and isNonnull($spritemap)}
		<svg aria-hidden="true" class="lexicon-icon lexicon-icon-{$symbol}{$elementClasses ? '  ' + $elementClasses : ''}">
			<use xlink:href="{$spritemap}#{$symbol}" />
		</svg>
	{elseif $elementClasses}
		<span aria-hidden="true" class="{$elementClasses}"></span>
	{/if}
{/template}