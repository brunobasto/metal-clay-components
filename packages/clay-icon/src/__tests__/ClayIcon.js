'use strict';

import dom from 'metal-dom';
import ClayIcon from '../ClayIcon';

const spritemap = '../node_modules/lexicon-ux/build/images/icons/icons.svg';

let clayIcon;

describe('ClayIcon', function() {
	afterEach(() => {
		if (clayIcon) {
			clayIcon.dispose();
		}
	});

	it('should generate markup for icon `add-cell`', function() {
		clayIcon = new ClayIcon({
			spritemap: spritemap,
			symbol: 'add-cell',
		});

		expect(clayIcon).toMatchSnapshot();
	});

	it('should append elementClasses to the outer element', function() {
		clayIcon = new ClayIcon({
			elementClasses: 'icon-monospaced',
			spritemap: spritemap,
			symbol: 'add-cell',
		});

		expect(dom.hasClass(clayIcon.element, 'icon-monospaced')).toBeTruthy();
	});

	it('should not output any markup if symbol is undefined', function() {
		clayIcon = new ClayIcon({
			spritemap: spritemap,
		});

		expect(clayIcon.element).toBeNull();
	});

	it('should not output any markup if spritemap is undefined', function() {
		clayIcon = new ClayIcon({
			symbol: 'add-cell',
		});

		expect(clayIcon.element).toBeNull();
	});

	it('should output markup for Glyphicons.', function() {
		clayIcon = new ClayIcon({
			elementClasses: 'glyphicon glyphicon-plus',
		});

		expect(clayIcon).toMatchSnapshot();
	});

	it('should output markup for Clay Font Awesome Icons.', function() {
		clayIcon = new ClayIcon({
			elementClasses: 'clay-fa icon-plus',
		});

		expect(clayIcon).toMatchSnapshot();
	});
});
