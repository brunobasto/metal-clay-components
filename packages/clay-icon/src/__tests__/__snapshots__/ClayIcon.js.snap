// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayIcon should generate markup for icon \`add-cell\` 1`] = `
<svg aria-hidden="true" class="lexicon-icon lexicon-icon-add-cell">
  <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#add-cell"></use>
</svg>
`;

exports[`ClayIcon should output markup for Clay Font Awesome Icons. 1`] = `<span aria-hidden="true" class="clay-fa icon-plus"></span>`;

exports[`ClayIcon should output markup for Glyphicons. 1`] = `<span aria-hidden="true" class="glyphicon glyphicon-plus"></span>`;
